File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k): 
    stack = []
    for i in range(len(number)):
        if stack and stack[-1] < number[i]:
            while stack and stack[-1] < number[i] and k > 0:
                stack.pop()
                k -= 1
            stack.append(number[i])
        
        else:
            stack.append(number[i])
    
    if k > 0:
        while k > 0:
            stack.pop()
            k -= 1
    
    return ''.join(stack)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var stack = [Character]()
    var k = k
    var answer = ""
    
    for num in Array(number){
        if !stack.isEmpty && stack.last! < num {
            // while stack 
            while !stack.isEmpty && stack.last! < num && k > 0 {
                stack.removeLast()
                k -= 1
            }
            stack.append(num)
        } else {
            stack.append(num)
        }
    }
    
    return String(stack.prefix(number.count - k))
}